components/eamxx/src/physics/nudging/eamxx_nudging_process_interface.cpp

```diff
@@ -124,19 +136,77 @@ void Nudging::apply_weighted_tendency(Field& base, const Field& next, const Fiel
 void Nudging::initialize_impl (const RunType /* run_type */)
 {
   using namespace ShortFieldTagsNames;
+  // Set up pointers for grids
+  // grid for coarse data
+  std::shared_ptr<scream::AbstractGrid> grid_ext;
+  // grid after vertical interpolation
+  std::shared_ptr<scream::AbstractGrid> grid_hxt;
+  // grid after horizontal interpolation
+  std::shared_ptr<scream::AbstractGrid> grid_int;
+
+  // Initialize the refining remapper stuff at the outset,
+  // because we need to know the grid information.
+  // For now, we are doing the horizontal interpolation last,
+  // so we use the m_grid (model physics) as the target grid
+  grid_int = m_grid->clone(m_grid->name(), true);
+  if (m_refine_remap) {
+    // P2P remapper
+    m_refine_remapper =
+        std::make_shared<RefiningRemapperP2P>(grid_int, m_refine_remap_file);
+    // If we are refine-remapping, then get grid from remapper
+    auto grid_ext_const = m_refine_remapper->get_src_grid();
+    // Deep clone it though to get rid of "const" stuff
+    grid_ext = grid_ext_const->clone(grid_ext_const->name(), true);
+    /* quick check here to ensure we are in good stnading */
+    // If the user gives a mapfile, we assume we are refine-remapping,
+    // but we should check if the mapfile is actually remapping stuff,
+    // so we compare the global columns of the target and source grids
+    auto grid_int_global_cols = grid_int->get_num_global_dofs();
+    auto grid_ext_global_cols = grid_ext->get_num_global_dofs();
+    EKAT_REQUIRE_MSG(
+        grid_int_global_cols != grid_ext_global_cols,
+        "Error! Nudging::initialize_impl - the mapfile given for "
+        "refine-remapping does not remap anything.  Please check the mapfile.");
```

Comment:
I think this is backwards, compared to what I had in mind. You set `m_refine_remap` if an entry is found in the input file specifying the map file and this line checks that indeed the two grids are different (no neX->neX remap).

However, I think the logic should be backwards:
1. if the nudging file has ncol different from eamxx phys grid, set m_refine_remap=true;
2. if refine_remap=true, you _must_ pass a map file, otherwise there should be _no_ map file provided. if map file provided when ncols matches, you could decide to error out or just issue a warning (it may be the user specified wrong nudging file, or maybe the nudging file is correct, and they forgot to delete the map file entry)
3. If a map file is specified, you should check that the two ncols match the nudging file and eamxx grid values.

Reply:
Uhm, no you can just do the checks in the `set_grids` method, when all grids are available. And you only need to check the map file if the model and nudging ncols are different (which you would have to do anyways, since you need to create a remapper).

Reply:
Notice that `set_grids` happens quite early in the init sequence. Atm procs creation is usually fast, and not much can happen in the constructor, since no grid info is yet available. I think it's perfectly ok to put this kind of checks in `set_grids`.